plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineSource.java

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import com.google.common.base.MoreObjects;
+
+import org.eclipse.che.api.core.model.machine.MachineSource;
+import org.eclipse.che.api.machine.server.model.impl.MachineSourceImpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.eclipse.che.plugin.docker.machine.DockerInstanceProvider.DOCKER_IMAGE_TYPE;
+
+/**
+ * Set of helper methods that identifies docker image properties
+ *
+ * @author Sergii Kabashnyuk
+ * @author Florent Benoit
+ */
+public class DockerMachineSource extends MachineSourceImpl {
```

Comment:
Sorry, but it doesn't make sense to me. I've offered to reuse existing component instead of creating Regexp and other logic yourself and test it. And you answered that it is not part of this issue.